### PR TITLE
fix: store fetched data shapes on steps, related cleanup

### LINKED ISSUE #1562: Fetch and store the data shapes when the user selects (or configures) the action on a connection
Currently we're fetching data shapes in various places from the backend, I think unnecessarily  The UI should just fetch the input/output data shapes if needed when the user selects the action and update the action descriptor on the step.

### PR DESCRIPTION
**Pull request by gashcrumb**
fixes #1562 

@igarashitm FYI

### CODE REVIEW

File: app/ui/src/app/integration/edit-page/step-configure/data-mapper/data-mapper-host.component.ts [hunk 1]
```diff
@@ -99,54 +99,20 @@ export class DataMapperHostComponent implements OnInit {
...
     const targetPair = this.currentFlowService.getSubsequentStepsWithDataShape(this.position)[1];
-    this.addInitializationTask();
-    this.currentFlowService.fetchInputDataShapeFor(targetPair.step).then(dataShape => {
-      if (!this.addTargetDocument(targetPair.step.id, targetPair.index, dataShape)) {
-        this.cfg.errorService.error(
-          'Unsupported target data type for Step "' + targetPair.step.name + '": ' + dataShape, '');
-        this.removeInitializationTask();
-        return;
-      }
-      this.currentFlowService.events.emit({
```
Comment (igarashitm) [inline app/ui/src/app/integration/edit-page/step-configure/data-mapper/data-mapper-host.component.ts]: I think we still need to fire this event, this is to copy data shapes into datamapper step from previous/subsequent steps.

Comment (gashcrumb) [reply app/ui/src/app/integration/edit-page/step-configure/data-mapper/data-mapper-host.component.ts]: ah, okay, I'll rework that, thanks!

**Comment by igarashitm** [comment]
Oops sorry, my DataShapeKinds enum work (#1568) introduced a bit conflict to yours in data mapper host...

**Comment by gashcrumb** [comment]
@igarashitm no worries, updated

**Pull request merged**